server/src/main/java/io/crate/execution/engine/distribution/merge/BatchPagingIterator.java

```diff
@@ -147,28 +148,18 @@ public boolean hasLazyResultSet() {
     }
 
     private void raiseIfClosedOrKilled() {
-        Throwable err;
-        synchronized (this) {
-            err = killed;
-        }
+        Throwable err = killed;
```

Comment:
does this help in any way, if `killed` gets re-assigned?

Reply:
Not sure what you mean with "safe" or "better".  `raiseIfClosedOrKilled` needs to observe the last `killed` value. This is the case with volatile.